common/logger/src/diem_logger.rs

```diff
@@ -296,7 +309,11 @@ impl DiemLogger {
 
     fn send_entry(&self, entry: LogEntry) {
         if let Some(printer) = &self.printer {
-            let s = format(&entry).expect("Unable to format");
+            let s = if let Some(formatter) = &self.custom_format {
+                formatter(&entry).expect("Unable to format")
+            } else {
+                format(&entry).expect("Unable to format")
+            };
```

Comment:
might be simpler if the `formatter` was `format` by default.

Reply:
Good call